File: Report-Analyzer/src/backend/database/models/state.ts
import { Schema, Document } from "mongoose";
import { ITestResult } from './testResult';
import { ITestResultContainer } from './testResultContainer';
export interface IState extends Document {
  TestResultId: ITestResult['_id']
  ContainerId: ITestResultContainer['_id']
  DerivationContainer: {[identifier: string] : string}
  DisplayName: string
  Result: string
  AdditionalResultInformation: string
  AdditionalTestInformation: string
  SrcPort: number
  DstPort: number
  StartTimestamp: string
  EndTimestamp: string
  uuid: string
  Stacktrace?: string
}

export const StateSchema = new Schema({
  TestResultId: {
    type: Schema.Types.ObjectId,
    ref: 'TestResult',
    required: true,
  },
  ContainerId: {
    type: Schema.Types.ObjectId,
    ref: 'TestContainer',
    required: true,
  },
  DerivationContainer: {
    type: Schema.Types.Map,
    of: String,
    default: new Map()
  },
  DisplayName: String,
  Result: String,
  AdditionalResultInformation: String,
  AdditionalTestInformation: String,
  SrcPort: Number,
  DstPort: Number,
  StartTimestamp: String,
  EndTimestamp: String,
  uuid: String,
  Stacktrace: String
})

StateSchema.index({TestResultId: 1})
StateSchema.index({ContainerId: 1})
StateSchema.index({Result: 1})
StateSchema.index({AdditionalResultInformation: 1})
StateSchema.index({Stacktrace: 1})

File: Report-Analyzer/src/backend/uploader/src/database/models/state.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"time"
)

type State struct {
	ID                          primitive.ObjectID `bson:"_id"`
	TestResultId                primitive.ObjectID `bson:"TestResultId"`
	ContainerId                 primitive.ObjectID `bson:"ContainerId"`
	DerivationContainer         map[string]string  `bson:"DerivationContainer"`
	DisplayName                 string             `bson:"DisplayName"`
	Result                      string             `bson:"Result"`
	AdditionalResultInformation string             `bson:"AdditionalResultInformation"`
	AdditionalTestInformation   string             `bson:"AdditionalTestInformation"`
	SrcPort                     int                `bson:"SrcPort"`
	DstPort                     int                `bson:"DstPort"`
	StartTimestamp              string             `bson:"StartTimestamp"`
	EndTimestamp                string             `bson:"EndTimestamp"`
	Uuid                        string             `bson:"uuid"`
	Stacktrace                  string             `bson:"Stacktrace"`
	CreatedAt                   time.Time          `bson:"createdAt"`
	UpdatedAt                   time.Time          `bson:"updatedAt"`
}
